tests/frontend/specs/nested_pad.js

```diff
@@ -0,0 +1,22 @@
+describe("Nested Pad", function(){
```

Comment:
this is no where near enough test coverage..  

Also where is ``:`` documented as a non allowed character in a padId?  What happens if I try to put ``foo:`` in as a padId?

etc. etc. etc.

Test coverage is very important, I expect to see a good 200 lines of as many edge cases as possible covered.  This here is not even touching the sides of what we'd need to merge such a potentially breaking feature!